packages/inertia/src/inertia.js

```diff
@@ -221,31 +278,33 @@ export default {
   },
 
   pushState(page) {
-    window.history.pushState({
-      cache: {},
-      ...page,
-    }, '', page.url)
+    this.page = page
+    window.history.pushState(page, '', page.url)
   },
 
-  replaceState(page, preserveState = false) {
-    window.history.replaceState({
-      ...{ cache: preserveState && window.history.state ? window.history.state.cache : {} },
-      ...page,
-    }, '', page.url)
+  replaceState(page) {
+    this.page = page
+    window.history.replaceState(page, '', page.url)
   },
 
-  restoreState(event) {
-    if (event.state) {
-      const page = this.transformProps(event.state)
+  handlePopstateEvent(event) {
+    if (event.state !== null) {
+      let page = event.state
       let visitId = this.createVisitId()
       return Promise.resolve(this.resolveComponent(page.component)).then(component => {
         if (visitId === this.visitId) {
+          this.page = page
           this.swapComponent({ component, page, preserveState: false }).then(() => {
-            this.restoreScrollPositions(page)
+            this.restoreScrollPositions()
             this.fireEvent('navigate', { detail: { page: page } })
           })
         }
       })
+    } else {
+      let url = this.normalizeUrl(this.page.url)
```

Comment:
Same as previous comment, you have no re-attribution, so const is better here 
```suggestion
      const url = this.normalizeUrl(this.page.url)
```